libs/platform/src/lib/components/form/checkbox/checkbox.component.ts

```diff
@@ -0,0 +1,196 @@
+import {
+    Component,
+    ChangeDetectorRef,
+    Self,
+    Optional,
+    EventEmitter,
+    Output,
+    Input,
+    ViewChild,
+    AfterViewInit,
+} from '@angular/core';
+import { NgControl, NgForm, FormControl } from '@angular/forms';
+import { CheckboxComponent as CoreCheckboxComponent } from '@fundamental-ngx/core';
+import { BaseInput } from '../base.input';
+import { Status } from './../form-control';
+
+@Component({
+    selector: 'fdp-checkbox',
+    templateUrl: './checkbox.component.html',
+})
+export class CheckboxComponent extends BaseInput implements AfterViewInit {
+    /** reference of child checkbox implementation */
+    @ViewChild(CoreCheckboxComponent)
+    private corecheckbox: CoreCheckboxComponent;
+
+    /** set to true if binary checkbox */
+    @Input()
+    isBinary: boolean = false;
+
+    /** Sets label for checkbox. */
+    @Input()
+    label: string;
+
+    /**
+     * Attached to the aria-label attribute of the host element. In most cases, aria-labelledby will
+     * take precedence so this may be omitted.
+     */
+    @Input('aria-label')
+    ariaLabel: string = '';
+
+    /**
+     * Users can specify the `aria-labelledby` attribute which will be forwarded to the input element
+     */
+    @Input('aria-labelledby')
+    ariaLabelledby: string | null = null;
+
+    /** indeterminate state */
+    @Input()
+    tristate: boolean = false;
+
+    /** indeterminate state can be selected  */
+    @Input()
+    tristateSelectable: boolean = true;
+
+    /** value for checkbox, when it is not binary */
+    @Input()
+    get value(): any {
+        return super.getValue();
+    }
+    set value(selectValue: any) {
+        this._value = selectValue;
+        this._cd.markForCheck();
```

Comment:
What allot of issue this is causing, so let;s fix those issues? 
What at least we need to do is compare if the _value is different than selectValue, only then set the value and fire markForCheck

Reply:
using setValue now. issues handled.
Done